client/l2tp_client.c

```diff
@@ -1449,46 +1484,43 @@ int main(int argc, char **argv)
       brokers[i].port);
 
     // Activate the broker.
-    main_context->standby_only = 0;
     main_context->state = STATE_GET_COOKIE;
 
     // Initially, we mark this broker as broken.  We will remove this mark after establishing
     // a connection.  We only want to consider a broker as broker if the initial connection fails;
     // disconnecting later (e.g. because the broker got restarted) is fine.
-    brokers[i].broken = 1;
+    brokers[i].broken = timer_now();
 
     // Perform processing on the main context; if the connection fails and does
-    // not recover after 30 seconds, restart the broker selection process.
-    int restart_timer = 0;
+    // not recover after 15 seconds, restart the broker selection process.
     time_t timer_establish = timer_now();
     for (;;) {
       broker_select_one(brokers[i]);
       context_process(main_context);
 
-      if (main_context->state == STATE_REINIT) {
+      if (main_context->state == STATE_FAILED) {
         syslog(LOG_ERR, "Connection to %s lost.", main_context->broker_hostname);
         break;
-      } else if (main_context->state == STATE_KEEPALIVE) {
-        // We successfully established a connection, this broker is fine.
-        brokers[i].broken = 0;
       }
 
       // If the connection is lost, we start the reconnection timer.
-      if (restart_timer && main_context->state != STATE_KEEPALIVE) {
+      // Hitting this code path should not be possible, but lets play safe.
```

Comment:
Can you add an explanation why in the comment?

Reply:
Done.